components/StaffView.tsx

```diff
@@ -22,38 +26,52 @@ const filters = [
   "Level 8",
 ];
 
-const StaffView: React.FC<StaffViewProp> = ({ staff }) => (
-  <div className={styles.compContainer}>
-    <div className={styles.leftContainer}>
-      <h1 className={styles.compTitle}>Staff</h1>
-      <div className={styles.compList}>
-        <ul className={styles.scroll}>
-          {staff.map((c) => (
-            <StaffCard
-              key={c.id}
-              firstName={c.firstName}
-              lastName={c.lastName}
-              phone_number={c.phoneNumber}
-              email={c.email}
-            />
+const StaffView: React.FC<StaffViewProp> = ({ staff }) => {
+  const client = useContext(APIContext);
+
+  // Use SWR hook to get the data from the backend
+  const { data, error } = useSWR("/api/staff", () => client.getStaff());
+
+  // Dynamically load the data from the backend and render the respective component
+  const staffData: any = (data: User[] | undefined, error: Error) => {
+    if (error) return <Error />;
+    if (!data) return <Loader />;
+
+    // Render the staff
+    return data.map((c: any) => (
+      <StaffCard
+        key={c.id}
+        firstName={c.firstName}
+        lastName={c.lastName}
+        phone_number={c.phoneNumber}
+        email={c.email}
+      />
+    ));
+  };
```

Comment:
Ok. Since the `any` in the function header is bothering me, I think its best to to split this into two components. It will mak e it easier once we have to add filtering and stuff to it. So, lets create a new component that looks something like this:

```javascript 
const StaffScroll: React.FC = () => {
  const client = useContext(APIContext);

  // Use SWR hook to get the data from the backend
  const { data, error } = useSWR("/api/staff", () => client.getStaff());

  if (error) return <><Error /></>
  if (!data) return <><Loader /></>;

  return (
    <>
      {
        data.map((c: any) => (
            <StaffCard
              key={c.id}
              firstName={c.firstName}
              lastName={c.lastName}
              phone_number={c.phoneNumber}
              email={c.email}
            />
        ))
      }
    </>
  )
}
```
Then, at line 57, all we have to do is call the `<StaffScroll\> component and things should just work :)

Reply:
Remove all of the duplicate code afterwards :)